File: edsdk/windows/models/xfile.cpp
#include "string"

class XFile{
    string videoPath;
    string imagePath;

public:
    XFile(){
        videoPath = "";
        imagePath = "";
    }

    XFile(string videoPath, string imagePath) : videoPath(videoPath), imagePath(imagePath){

    }

    std::string toJson() const {
        std::string jsonStr = "{";
        jsonStr += "\"videoPath\":\"" + videoPath + "\",";
        jsonStr += "\"imagePath\":\"" + imagePath + "\"";
        jsonStr += "}";
        return jsonStr;
    }
};

File: edsdk/lib/models/xfile.dart
class XFile {
  final String imagePath;
  final String videoPath;

  XFile(this.imagePath, this.videoPath);

  static XFile fromJson(Map<String, dynamic> json) {
    return XFile(
      json["imagePath"] as String,
      json["videoPath"] as String,
    );
  }

  @override
  String toString() {
    return "XFile(imagePath: $imagePath,videoPath: $videoPath)";
  }
}
